From d621ac52a4244bc53123f5e89acc90d4644ecf19 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo.salveti@linaro.org>
Date: Tue, 15 Nov 2016 21:35:17 -0200
Subject: [PATCH] net/mlx4_core: enable enable_4k_uar by default

CentOS uses 64KB pagesize on ARM64, so the driver will always fail to
load by default, unless enable_4k_uar gets set by the user. If there
are compatibility issues with old VF drivers, the user can still disable
this setting by setting it to false.

Signed-off-by: Ricardo Salveti <ricardo.salveti@linaro.org>
---
 drivers/net/ethernet/mellanox/mlx4/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/mellanox/mlx4/main.c b/drivers/net/ethernet/mellanox/mlx4/main.c
index 6f4e67b..e33b480 100644
--- a/drivers/net/ethernet/mellanox/mlx4/main.c
+++ b/drivers/net/ethernet/mellanox/mlx4/main.c
@@ -105,10 +105,10 @@ module_param(enable_64b_cqe_eqe, bool, 0444);
 MODULE_PARM_DESC(enable_64b_cqe_eqe,
 		 "Enable 64 byte CQEs/EQEs when the FW supports this (default: True)");
 
-static bool enable_4k_uar;
+static bool enable_4k_uar = true;
 module_param(enable_4k_uar, bool, 0444);
 MODULE_PARM_DESC(enable_4k_uar,
-		 "Enable using 4K UAR. Should not be enabled if have VFs which do not support 4K UARs (default: false)");
+		 "Enable using 4K UAR. Should not be enabled if have VFs which do not support 4K UARs (default: True)");
 
 #define PF_CONTEXT_BEHAVIOUR_MASK	(MLX4_FUNC_CAP_64B_EQE_CQE | \
 					 MLX4_FUNC_CAP_EQE_CQE_STRIDE | \
-- 
2.7.4

